server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1695,131 +1596,235 @@ protected void setConfigProperties() throws ConfigurationException, IOException,
-        if (started && !quiet) {
-            log.info("Go to {}", this::getURL);
-        }
-        return started;
-    }
-
-    /**
-     * @see #stop(boolean)
-     */
-    public void stop() {
-        stop(false);
-    }
-
     /**
      * Call {@link #doStart(boolean)} with false as parameter.
      *
      * @see #doStart(boolean)
      * @return true if the server started successfully
      */
-    public boolean doStart() throws PackageException {
-        boolean started = doStart(false);
-        if (started && !quiet) {
-            log.info("Go to {}", this::getURL);
+    public boolean doStart() {
+        if (doStart(false).isAlive()) {
+            if (!quiet) {
+                log.info("Go to {}", this::getURL);
+            }
+            return true;
         }
-        return started;
+        return false;
     }
 
     /**
-     * Whereas {@link #doStart()} considers the server as started when the process is running, {@link #doStartAndWait()}
-     * waits for effective start by watching the logs
+     * Whereas {@link #doStart()} considers the server as started when the process is running, this method waits for
+     * effective start by calling {@link #statusServletClient}.
      *
-     * @param logProcessOutput Must process output stream must be logged or not.
      * @return true if the server started successfully
      */
-    public boolean doStartAndWait(boolean logProcessOutput) throws PackageException {
-        boolean commandSucceeded = false;
-        if (doStart(logProcessOutput)) {
-            addShutdownHook();
-            try {
-                if (waitForEffectiveStart()) {
-                    commandSucceeded = true;
+    public boolean doStartAndWait() {
+        var nuxeoProcess = doStart(false);
+        if (nuxeoProcess.isAlive()) {
+            //noinspection unused
+            try (var hook = new ShutdownHook(this)) { // NOSONAR try-with-resources
+                waitForEffectiveStart(nuxeoProcess);
+                if (!quiet) {
+                    log.info("Go to {}", this::getURL);
                 }
-                removeShutdownHook();
+                return true;
             } catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
                 throw new RuntimeException(e);
             }
         }
-        return commandSucceeded;
+        return false;
+    }
+
+    /**
+     * @since 11.1
+     */
+    public void doConsole() {
+        var hook = new ShutdownHook(this);
+        try {
+            doStart(true).onExit().thenAccept(p -> {
+                hook.close();
+                // always error, nuxeo process is not supposed to exit
+                System.exit(EXIT_CODE_ERROR);
+            });
+        } catch (RuntimeException e) {
+            // for errors in doStart
+            hook.close();
+            throw e;
+        }
+        if (!quiet) {
+            log.info("Go to {}", this::getURL);
```

Comment:
This line appears 3 times, don't know if it'interesting to refactor?

Reply:
I can write a method, I didn't do it because it would introduce more line than the one we're refactoring. Wdyt?

Reply:
I think it's OK like this.